### PR TITLE
Show result of comparing base branch with master on pending tests

### PR DESCRIPTION
**Pull request by linrock**
@vondele this is an update to the "Master diff" link in the actions sidebar section for pending tests as you suggested here: https://github.com/glinscott/fishtest/pull/558#issuecomment-610022442

The client will check the diff between the base branch of the submitted test and official Stockfish master. If there's no difference, it shows a green success message. Otherwise, a red error message.

---
If there's no difference between the base branch and official Stockfish master:

---
If the base branch doesn't match official Stockfish master:

### CODE REVIEW

**Comment by vondele** [comment]
Looks excellent!

(not a review, I notice one strange piece of syntax).

File: fishtest/fishtest/templates/tests_view.mak [hunk 1]
```diff
@@ -209,3 +217,24 @@ Gaussian Kernel Smoother&nbsp;&nbsp;<div class="btn-group"><button id="btn_smoot
...
+  $(function() {
+    var apiUrlBaseMaster = "https://api.github.com/repos/official-stockfish/Stockfish";
+    $.ajax({
+      url: apiUrlBaseMaster + "/compare/master...${run['args']['resolved_base'][:7]}",
+      headers: {
+        Accept: "application/vnd.github.v3.diff"
+      },
+      success: function(response) {
+        var $masterDiff = $("#master-diff");
+        if (response.length === 0) {
```
Comment (vondele) [inline fishtest/fishtest/templates/tests_view.mak:231]: is  `response.length === 0` (3x =) correct syntax ?

Comment (tomtor) [reply fishtest/fishtest/templates/tests_view.mak:231]: Yes, it's equal without type coercion.

**Comment by ppigazzini** [comment]
@linrock please rebase on new master

**Comment by tomtor** [comment]
Hi @linrock Can you also add your name to AUTHORS in this commit?

**Comment by ppigazzini** [comment]
@tomtor @linrock #560 AUTHORS already updated :)

**Comment by ppigazzini** [comment]
@vondele @tomtor CI checks stopped working?

**Comment by tomtor** [comment]
> @vondele @tomtor CI checks stopped working?

Yes and no: You can see them here:

https://travis-ci.org/github/glinscott/fishtest/builds

Not sure what to change to recreate the green/red markers in the overview

**Comment by vondele** [comment]
Several projects have had this problem, it is unclear how to solve it. Disabling and reabling the github integration worked for some, not for all.

**Comment by ppigazzini** [comment]
@vondele the SF Readme.md references the @mcostalba repo: is it correct? 
https://raw.githubusercontent.com/official-stockfish/Stockfish/master/Readme.md
```
## Overview

[![Build Status](https://travis-ci.org/official-stockfish/Stockfish.svg?branch=master)](https://travis-ci.org/official-stockfish/Stockfish)
[![Build Status](https://ci.appveyor.com/api/projects/status/github/official-stockfish/Stockfish?branch=master&svg=true)](https://ci.appveyor.com/project/mcostalba/stockfish/branch/master)

```

**Comment by vondele** [comment]
hmm seems like it works, but it is possibly not the best setup.

**Comment by linrock** [comment]
@ppigazzini i've rebased on master and squashed into one commit. ready for testing!

**Comment by ppigazzini** [comment]
DEV updated

**Comment by ppigazzini** [comment]
Looks good to me.
@tomtor perhaps we should write a different message in "Actions" when approving with the Orange button (base branch different from official master)

**Comment by vondele** [comment]
Dev seems to work as expected... not sure it is very important to adjust Actions. If it is trivial, why not, otherwise I would leave as is. The most important part if to give an easy indication to the approver, all info is there somehow anyway.

**Comment by linrock** [comment]
Since it's unsafe to click "Approve" when the base branch is different from official master, in these cases I suggest that the "Approve" button is disabled until the approver clicks on the "Master diff" link. Adding that extra step ensures the approver looks at the diff before they can click the button.

If this behavior sounds good, i can open a separate PR for it.

**Comment by ppigazzini** [comment]
PROD updated.
@linrock you proposal looks good to me :)

**Comment by vondele** [comment]
I'm not sure a requirement is needed. In some cases, e.g. long term devs that trigger a regtest, looking at the diff makes no sense (1000s of lines), and we kind of need to trust the dev. 

Is it possible to check if the selected sha is the same as an existing one if official-stockfish first, and skip the requirement if the sha corresponds to a commit in official stockfish? If they are the same, it should not be a requirement to check (the code won't be malicious). Regression tests, and devs that are a commit or so behind are relatively common.

The other case where large diffs can exist is for the tunes, as people have to merge the tuning branch. These diffs are practically also un-checkable. Not sure if we have a good solution (one option would be to integrate the tuning branch in master).

**Comment by ppigazzini** [comment]
@vondele using a previous commit as base makes a green "Approve" button, tested with this
https://dfts-0.pigazzini.it/tests/view/5e8c9f1918ffe8a1c9794f27

To get an orange "Approve" button I merged a new commit (new txt file) in the master.

**Comment by ppigazzini** [comment]
@linrock mmm, it seems that we have a problem with old tests, view at example https://tests.stockfishchess.org/tests/view/5214a8390ebc59160a0182f6
this seems the last test OK
https://tests.stockfishchess.org/tests/view/5214ce9a0ebc59160a0182f8

**Comment by linrock** [comment]
@ppigazzini got it, i'll look into fixing it. seems like for older tests we will have to check if both the test is not finished + not approved to decide whether to show that message.

**Pull request merged**